### ISSUE TITLE
One update updates the values multiple times | Halloween Problem

### ISSUE THREAD

**Issue by latelatif** [cmu-db/peloton#1222]
![pasted image at 2018_03_23 12_02 pm](https://user-images.githubusercontent.com/25693934/37872182-2f673130-2fd0-11e8-8ab8-7f5bd217ddd8.png)

As can be seen in the trace, the updates happen more than once. This looks like the Halloween problem (https://en.wikipedia.org/wiki/Halloween_Problem).

This behavior is only seen with codegen and not the old engine.

**Comment by pmenon** [comment]
I recall running into this last summer and fixing it. But right now, I can't reproduce this on master:

```
pmenon=# create database pmenon;
CREATE DATABASE
pmenon=# create table foo (id integer, year integer);
CREATE TABLE
pmenon=# insert into foo values (10, 1000);
INSERT 0 1
pmenon=# select * from foo;
 id | year 
----+------
 10 | 1000
(1 row)

pmenon=# update foo set id = id / 2;
UPDATE 1
pmenon=# select * from foo;
 id | year 
----+------
  5 | 1000
(1 row)

pmenon=# update foo set year = year / 2;
UPDATE 1
pmenon=# select * from foo;
 id | year 
----+------
  5 |  500
(1 row)

pmenon=# 
```
Can provide more details on build environment? OS, compiler, cmake settings?

**Comment by pervazea** [comment]
I'm seeing this behavior w/o codegen too.

**Comment by pervazea** [comment]
@pmenon

**Comment by poojanilangekar** [comment]
Fixed in #1272 and #1263
